trillian_log_api.proto

```diff
@@ -219,60 +235,83 @@ message GetLeavesByHashRequest {
 }
 
 message GetLeavesByHashResponse {
-    // TODO(gbelvin) reply with error codes.
+    // TODO(gbelvin) reply with error codes. Reuse QueueLogLeaf?
     repeated LogLeaf leaves = 2;
 }
 
-// QueuedLogLeaf represents a log leaf that has been queued for inclusion; it may
-// be pending or already exist in the log (if the log does not allow duplicates).
+// A result of submitting an entry to the log. Output only.
+//
+// TODO(pavelkalinnikov): Consider renaming it to AddLogLeafResult or the like.
+// It can be reused by AddSequencedLeaves API.
 message QueuedLogLeaf {
-    // The leaf is present if status.code is:
-    //  - google.rpc.OK : the leaf is the same as in the QueueLea{f,ves}Request
-    //  - google.rpc.ALREADY_EXISTS : the leaf is the one already present in the log.
+    // The leaf as it was stored by Trillian. Empty unless `status.code` is:
+    //  - `google.rpc.OK`: the `leaf` data is the same as in the request.
+    //  - `google.rpc.ALREADY_EXISTS` or 'google.rpc.FAILED_PRECONDITION`: the
+    //    `leaf` is the conflicting one already in the log.
     LogLeaf leaf = 1;
+
+    // The status of adding the leaf.
+    //  - `google.rpc.OK`: successfully added.
+    //  - `google.rpc.ALREADY_EXISTS`: the leaf is a duplicate of an already
+    //    existing one. Either `leaf_identity_hash` is the same in the `LOG`
+    //    mode, or `leaf_index` in the `PREORDERED_LOG`.
+    //  - `google.rpc.FAILED_PRECONDITION`: A conflicting entry is already
+    //    present in the log, e.g., same `leaf_index` but different `leaf_data`.
     google.rpc.Status status = 2;
 }
 
+// A leaf of the log's Merkle tree, corresponds to a single log entry. Each leaf
+// has a unique `leaf_index` in the scope of this tree.
 message LogLeaf {
-    // merkle_leaf_hash is over leaf data and optional extra_data.
+    // Output only. The hash over `leaf_data`.
```

Comment:
Apparently, `extra_data` is [not included](https://github.com/google/trillian/blob/7185394727abf33444773cc1bacd6a5d7c98ab11/server/log_rpc_server.go#L105) in the hash. Should it be?

Reply:
No, this is just a way of associating informational metadata with the entry at submit time. If a personality wanted to commit to that data then, as you say, they'd make it part of the `leaf_value`.